// The Firmament project
// Copyright (c) 2013 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// Represents a scheduling change to be applied.

syntax = "proto3";

package firmament;

message SchedulingDelta {
  enum ChangeType {
    PLACE = 0;
    PREEMPT = 1;
    MIGRATE = 2;
    NOOP = 3;
  }

  uint64 task_id = 1;
  string resource_id = 2;
  ChangeType type = 3;
}
